streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableMapValues.java

```diff
@@ -44,19 +45,24 @@ public KTableMapValues(final KTableImpl<K, ?, V> parent, final ValueMapperWithKe
 
     @Override
     public KTableValueGetterSupplier<K, V1> view() {
-        final KTableValueGetterSupplier<K, V> parentValueGetterSupplier = parent.valueGetterSupplier();
-
-        return new KTableValueGetterSupplier<K, V1>() {
-
-            public KTableValueGetter<K, V1> get() {
-                return new KTableMapValuesValueGetter(parentValueGetterSupplier.get());
-            }
-
-            @Override
-            public String[] storeNames() {
-                return parentValueGetterSupplier.storeNames();
-            }
-        };
+        // if the KTable is materialized, use the materialized store to return getter value;
```

Comment:
This looks indentical to the same in `KTableFilter` - can we not have another class that they both call that does this?

Reply:
@mjsax has the same comment, and I replied here: https://github.com/apache/kafka/pull/4421#discussion_r162220745